#!/usr/bin/env python
PACKAGE = "mppi_omav_interaction"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

######################
### General Config ###
######################

gen.add("reset", bool_t, 0, "Reset Position", False)

######################
### Goal Positions ###
######################

gen.add("ref_pos_x", double_t, 0, "Reference Position x", 0.0, -5.0, 5.0)
gen.add("ref_pos_y", double_t, 0, "Reference Position y", 0.0, -5.0, 5.0)
gen.add("ref_pos_z", double_t, 0, "Reference Position z", 1.0, 0.0, 2.0)
gen.add("ref_roll", double_t, 0, "Reference Roll", 0.0, -180.0, 180)
gen.add("ref_pitch", double_t, 0, "Reference Pitch", 0.0, -180.0, 180)
gen.add("ref_yaw", double_t, 0, "Reference Yaw", 0.0, -180.0, 180)

exit(gen.generate(PACKAGE, "omav_interaction_goal", "MPPIOmavReference"))
